File: dev-docs/snippets/events/cache-update-failed.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Fires when an internal KV cache entry fails to refresh from the server.
export function cacheUpdateFailed(client: JsBaoClient) {
  // #region example
  // Like `cacheUpdated`, this event isn't in the typed event map — subscribe via an untyped cast.
  type CacheUpdateFailed = { key: string; error: string };
  const on = client.on as (type: string, f: (payload: CacheUpdateFailed) => void) => void;
  on("cacheUpdateFailed", (payload) => {
    console.log("cache refresh failed:", payload.key, payload.error);
  });
  // #endregion example
}

File: dev-docs/snippets/events/cache-update-failed.swift
import JsBaoClient

// Fires when an internal KV cache entry fails to refresh from the server.
func onCacheUpdateFailed(client: JsBaoClient) {
  // #region example
  let sub = client.events.on(.cacheUpdateFailed) { (event: CacheUpdateFailedEvent) in
    print("cache refresh failed for", event.key, ":", event.error)
  }
  // #endregion example
  _ = sub
}
